### DISCUSSION TITLE
Thoughts on the cross-domain problem of images

### DISCUSSION THREAD

**Discussion by lrbmike** [flet-dev/flet#3900] [General]
I studied streamlit before, and when I wanted to migrate the function to flet, I found that the image had a cross-domain problem, and the same image address would not appear cross-domain in streamlit, but it would appear in flet, which made me very confused.

streamlit code:

![微信截图_20240830222902](https://github.com/user-attachments/assets/50422074-3870-44bf-b3c6-228643dd752d)

flet error:

![微信截图_20240830223120](https://github.com/user-attachments/assets/f6e679d2-ea63-4416-844e-088befd2ab98)

As far as I know, streamlit seems to handle cross-domain images through internal servers, so does flet have a similar processing mechanism? If not, can you refer to this method?

And now I'm using the FLASK framework to open a proxy server inside the flet to handle the cross-domain problem of images，the code is as follows
```python
# proxy server
from flask import Flask, request, send_file
import threading
import requests
from io import BytesIO

# create Flask app
app = Flask(__name__)


# proxy router
@app.route('/proxy')
def proxy():
    url = request.args.get('url')
    response = requests.get(url)
    return send_file(BytesIO(response.content), mimetype='image/jpeg')


# run Flask thread
def run_flask():
    app.run(port=5000)


flask_thread = threading.Thread(target=run_flask)
flask_thread.daemon = True
flask_thread.start()
````

image proxy
```python
image_url = "http://localhost:5000/proxy?url=" + image_url 
```

**Comment by ndonkoHenri** [comment]
Try using the html web renderer:
```py
ft.app(main, web_renderer=ft.WebRenderer.HTML)
```

**Reply by lrbmike** [reply]
If I run as a web app  (flet run --web main.py), it work. But if not, it doesn't.
So if I want to run as a desktop app, Is there any way to deal with it ?
